cwms-data-api/src/test/java/cwms/cda/data/dto/forecast/ForecastSpecTest.java

```diff
@@ -0,0 +1,97 @@
+package cwms.cda.data.dto.forecast;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import cwms.cda.api.DataApiTestIT;
+import cwms.cda.formatters.json.JsonV2;
+import org.apache.commons.io.IOUtils;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ForecastSpecTest {
+
+  @Test
+  void testRoundTripJson() throws JsonProcessingException
+  {
+    ForecastSpec s1 = buildForecastSpec();
+
+    ObjectMapper om = buildObjectMapper();
+
+    String jsonString = om.writeValueAsString(s1);
+    assertNotNull(jsonString);
+
+    ForecastSpec s2 = om.readValue(jsonString, ForecastSpec.class);
+    assertNotNull(s2);
+
+    assertForecastSpecEquals(s1, s2);
+  }
+
+  @Test
+  void testRoundTripEmptyJson() throws JsonProcessingException
+  {
+    ForecastSpec s1 = buildEmptyForecastSpec();
+
+    ObjectMapper om = buildObjectMapper();
+
+    String jsonString = om.writeValueAsString(s1);
+    assertNotNull(jsonString);
+
+    ForecastSpec s2 = om.readValue(jsonString, ForecastSpec.class);
+    assertNotNull(s2);
+
+    assertForecastSpecEquals(s1, s2);
+  }
+
+  @Test
+  void testJsonFile() throws IOException {
+    String json;
+    try (InputStream stream = getClass().getResourceAsStream("forecast_spec_test.json")) {
+      assertNotNull(stream);
+      json = IOUtils.toString(stream, StandardCharsets.UTF_8);
+    }
+
+    ObjectMapper om = buildObjectMapper();
+    ForecastSpec fi = om.readValue(json, ForecastSpec.class);
+
+    assertNotNull(fi);
+    assertForecastSpecEquals(fi, buildForecastSpec());
+  }
+
+  @NotNull
+  private ForecastSpec buildForecastSpec() {
+    ArrayList<String> timeSeriesIds = new ArrayList<>();
+    timeSeriesIds.add("tsid1");
+    timeSeriesIds.add("tsid2");
+    timeSeriesIds.add("tsid3");
+
+    return new ForecastSpec("spec", "office", "location", "sourceEntity",
+            "description", timeSeriesIds);
+  }
+
+  @NotNull
+  private ForecastSpec buildEmptyForecastSpec() {
+    return new ForecastSpec(null, null, null, null, null, null);
+  }
+
+  @NotNull
+  public static ObjectMapper buildObjectMapper() {
+    return JsonV2.buildObjectMapper();
+  }
+
+  void assertForecastSpecEquals(ForecastSpec s1, ForecastSpec s2) {
```

Comment:
I'd update this to assert that the generated string is equal, that way you don't have to keep adding to this method every time a field is added/edited/removed

Reply:
Are the JSON object guaranteed to have their keys in the same order? Also I modified the whitespace in the test files for readability

Reply:
Oh serialized/generated string do you not mean the JSON string?

Reply:
yeah I mean the JSON string, I think the order is consistent, though not dependable? if it isn't you can always add
```
@JsonPropertyOrder(alphabetic = true)
```
or
```
@JsonPropertyOrder({ "id", "label", "target", "source", "attributes" })
```

Reply:
Looks like it works without needing to define the order!